crates/ekore/src/anomalous_dimensions/unpolarized/spacelike/as4.rs

```diff
@@ -0,0 +1,58 @@
+//! |N3LO| |QCD|.
+
+pub mod ggg;
+pub mod ggq;
+pub mod gnsm;
+pub mod gnsp;
+pub mod gnsv;
+pub mod gps;
+pub mod gqg;
+
+use crate::harmonics::cache::Cache;
+use num::complex::Complex;
+
+// Compute the singlet anomalous dimension matrix.
+pub fn gamma_singlet(c: &mut Cache, nf: u8, variation: [u8; 4]) -> [[Complex<f64>; 2]; 2] {
+    let gamma_qq = gnsp::gamma_nsp(c, nf, variation[3]) + gps::gamma_ps(c, nf, variation[3]);
+    [
+        [gamma_qq, gqg::gamma_qg(c, nf, variation[2])],
+        [
+            ggq::gamma_gq(c, nf, variation[1]),
+            ggg::gamma_gg(c, nf, variation[0]),
+        ],
+    ]
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use crate::harmonics::cache::Cache;
+    use crate::{assert_approx_eq_cmplx, cmplx};
+    use num::complex::Complex;
+
+    #[test]
+    fn test_momentum_conservation() {
+        let NF = 5;
+        let mut c = Cache::new(cmplx!(2., 0.));
+        let quark_refs: [f64; 3] = [0.053441, 0.225674, -0.118792];
+        let gluon_refs: [f64; 3] = [-0.0300842, 0.283004, -0.343172];
```

Comment:
1. can you please add a reference? (python is acceptable)
2. shouldn't they be 0?

Reply:
Added a reference to python, I guess that is the precision of the approximations ?
(I think to have checked the numbers also in Mathematica...)

Reply:
they are non-zero also at $a_s^3$ so fine I guess?